src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1286,20 +1286,41 @@ private async ValueTask<(Stream?, TransportContext?, HttpResponseMessage?)> Conn
             }
         }
 
-        private static readonly SocketsConnectionFactory s_defaultConnectionFactory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
+        private static async ValueTask<Stream> DefaultConnectAsync(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
+        {
+            Socket socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
+            socket.NoDelay = true;
+            socket.DualMode = true;
```

Comment:
What happens if the OS doesn't support IPv6? I'm wondering if this should instead use the 2-arg ctor:
```
var socket = new Socket(SocketType.Stream, ProtocolType.Tcp) { NoDelay = true };
```

Reply:
This is just transferring over the existing code from SocketsConnectionFactory. But I agree, it seems a bit questionable.

@scalablecory Do you know why SocketsConnectionFactory was doing this?

Reply:
I'm going to change this as @stephentoub suggested. @scalablecory if you have any thoughts here, let me know.

Reply:
was likely a bug.